Assets/Scripts/Models/RoomManager.cs

```diff
@@ -0,0 +1,320 @@
+﻿using System;
+using System.Collections.Generic;
+
+public class RoomManager
+{
+    public event Action<Room> Adding;
+
+    public event Action<Room> Added;
+
+    public event Action<Room, Room> Split;
+
+    public event Action<Room, Room> Joined;
+
+    public event Action<Room> Removing;
+
+    public event Action<Room> Removed;
+
+    public List<Room> rooms;
```

Comment:
it is needed in other places, and I didn't want to make a GetRooms function as it would either copy all the data(expensive-ish) or do the same thing...